### ISSUE TITLE
forEach controller

### ISSUE THREAD

**Issue by kirillyu** [abstracta/jmeter-java-dsl#65]
There are many cases when you should to use jsonExtractor with match No. param setted to "-1". In this case you will get the array, which seems like:
var_1="val"
var_2="val2"
var_3="val3"
var_matchNr=3

In current DSL implementations no way to easy and usefull work with it. The best choice is to use the forEach controller, which can get only "var" parameter and iterate in with setting local variable for loop.

**Comment by rabelenda** [comment]
Thank you for reporting this, we will consider it for a future version since it would greatly simplify such use cases.

Until a new version is released you can use a while controller like this:

```java
testPlan(
        threadGroup(1, 1,
            jsr223Sampler("'[1, 2, 3]'")
                .children(
                    jsonExtractor("ids", "[]")
                        .matchNumber(-1)
                ),
            whileController(
                "${__groovy(vars.getObject('__jm__while__idx') < Integer.parseInt(vars.get('ids_matchNr')))}",
                jsr223Sampler("println vars['ids_' + (vars.getObject('__jm__while__idx') + 1)]")
            )
        )
    ).run();
```

Regards

**Comment by kirillyu** [comment]
Exactly what I did)

**Comment by kirillyu** [comment]
forEach released in[ 0.44 version ](https://github.com/abstracta/jmeter-java-dsl/releases/tag/v0.44)
